### PR TITLE
Bcmohad 15790 v1

### PR DESCRIPTION
**Pull request by anudishCGI**
Removal of Date ID population from trigger
Using existing flow to populate Date ID field for records coming from Integration
Could you please review @cgijeffolsen  @crystalz2022  @rahulmantriSFDC  @NataliaNikishina 

Thank You

### CODE REVIEW

File: TPL/force-app/main/default/flows/Update_Health_Care_Cost_Fields_On_Create.flow-meta.xml [hunk 1]
```diff
@@ -448,6 +455,11 @@
     <description>This is the subflow which would handle the On Create event on Health Care Cost Object.
 Update to PHN Value reference</description>
     <environments>Default</environments>
+    <formulas>
+        <name>dateidformula</name>
+        <dataType>String</dataType>
+        <expression>SUBSTITUTE(if(Text({!LoopThroughHealthCareCost.Date_of_Service__c})==null,&apos;00000000&apos;,Text({!LoopThroughHealthCareCost.Date_of_Service__c})), &apos;-&apos;, &apos;&apos;)&amp;&apos;&apos;&amp;{!LoopThroughHealthCareCost.Id}</expression>
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/flows/Update_Health_Care_Cost_Fields_On_Create.flow-meta.xml:461]: The application of a dateId formula is a good approach to update this item inline. The default 00000000 is correct for when Date of Service is empty.

File: TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls [hunk 2]
```diff
@@ -47,26 +47,4 @@ public with sharing class HealthcareCostTriggerHandler {
             System.debug(e.getMessage());
         }
     }
-    public static void populateDateId(List<Healthcare_Cost__c> hccList){
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls:50]: Removal of this trigger action puts all the Healthcare cost updates into the trigger where they can be managed in one place.

File: TPL/force-app/main/default/flows/Update_Health_Care_Cost_Fields_On_Create.flow-meta.xml [hunk 3]
```diff
@@ -188,6 +188,13 @@
                 <elementReference>LoopThroughHealthCareCost.Id</elementReference>
             </value>
         </assignmentItems>
+        <assignmentItems>
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/flows/Update_Health_Care_Cost_Fields_On_Create.flow-meta.xml:191]: This is the actual assignment, which is done universally on all healthcare cost records.

**Review by cgijeffolsen — APPROVED** [review]
@anudishCGI Looks good, approved.

**Review by rahul-mantri — APPROVED** [review]
Approved @anudishCGI 

CC: @NataliaNikishina @crystalz2022

**Review by NataliaNikishina — APPROVED** [review]
@anudishCGI approved.
cc: @crystalz2022

**Review by crystalz2022 — APPROVED** [review]
approved @anudishCGI

**Pull request merged**